docs/source/python/ipc.rst

```diff
@@ -89,10 +88,11 @@ convenience function ``pyarrow.ipc.open_stream``:
 
 .. ipython:: python
 
-   reader = pa.ipc.open_stream(buf)
-   reader.schema
-
-   batches = [b for b in reader]
+   with pa.ipc.open_stream(buf) as reader:
+         schema = reader.schema
+         batches = [b for b in reader]
```

Comment:
Took me a while to get around this one, if you wonder why this block is indented to more spaces than the other blocks, it's to work around what seemed like a bug in ipython directive.
When indented normally the second line was parsed as outside of the `with` block
```
In [0]: with pa.ipc.open_stream(buf) as reader:
   ...:    schema = reader.schema
In [1]: batches = [b for b in reader]
```

Reply:
Uh.